### ISSUE TITLE
[FEATURE] scale hyperparameter for the RepeatingBasisFunction

### ISSUE THREAD

**Issue by Garve** [koaning/scikit-lego#446]
Hi!

How about a hyperparameter that controls the width of the RBFs?
You set it to a (sensible) fixed value so far.https://github.com/koaning/scikit-lego/blob/51ef0c7f4fb0ac8717b3660931b78633831e170e/sklego/preprocessing/repeatingbasis.py#L86

But how about some `scale` hyperparameter and changing this line to
```python
self.width_ = self.scale / self.n_periods
```
with a default parameter of 1? I think it's good to have a choice there.

Best
Robert

**Comment by koaning** [comment]
Considering this is a preprocessing step, would this really matter? The idea is that the model that follows would scale the feature, no?

**Comment by MBrouns** [comment]
I think @Garve means that the `scale_` feature should influence the width of the generated rbf features. Increasing the width would make farther away values in the original space more alike in the transformed space and therefore influence the model fitting.

Adding it makes sense to me, but I understand the confusion with calling it `scale`.

**Comment by Garve** [comment]
Yeah, the model (let's say linear regression) would scale the bumps along the y-axis, but the width (i.e. some kind of a standard deviation) can't be changed by the model.

Let's take this example: 
```python
from sklego.preprocessing import RepeatingBasisFunction
from sklearn.linear_model import LinearRegression
from sklearn.pipeline import make_pipeline
import numpy as np
import matplotlib.pyplot as plt

X = np.linspace(-10, 10, 100)
y = np.exp(-X**2)
X = X.reshape(-1, 1)

pipe = make_pipeline(RepeatingBasisFunction(n_periods=12), LinearRegression(positive=True, fit_intercept=False))
pipe.fit(X, y)
plt.plot(X, y, label='Data')
plt.plot(X, pipe.predict(X), label='Prediction')
plt.legend()
```

![image](https://user-images.githubusercontent.com/932327/110235906-15c19580-7f33-11eb-9971-d0e3d2287eb3.png)

The model has a hard time fitting this simple function, which is only due to the fixed width. Setting `n_periods=24` gives the model the ability to learn it better, but then it needs 3 bumps to do so.

![image](https://user-images.githubusercontent.com/932327/110235941-5a4d3100-7f33-11eb-89da-9490776734b5.png)

The regression coefficients:
```python 
plt.bar(range(24), pipe.named_steps['linearregression'].coef_)
```

![image](https://user-images.githubusercontent.com/932327/110235965-8072d100-7f33-11eb-927e-a65fff22ec01.png)

Thus, the model loses its often desired sparsity.

**Comment by Garve** [comment]
> I think @Garve means that the `scale_` feature should influence the width of the generated rbf features. Increasing the width would make farther away values in the original space more alike in the transformed space and therefore influence the model fitting.
> 
> Adding it makes sense to me, but I understand the confusion with calling it `scale`.

Oh yeah, I mean scale in the sense of being able to *scaling the width* of the bumps. Sure, the name can be changed to something else, I just didn't want to call it width again since there is `width_` already. And this shouldn't change much, there should be just some scaling factor to it.

**Comment by koaning** [comment]
I'm open to the extra term as long as we keep the default as is.

**Comment by Garve** [comment]
Sure, I would also keep the default like this. It usually works quite well and 1 is a great constant, nobody asks questions ;)
